### PR TITLE
Cleanup for PR 369

### PR DESCRIPTION
**Pull request by lestrrat**

### CODE REVIEW

File: filter.go [hunk 1]
```diff
@@ -215,50 +217,63 @@ var filterBufPool = sync.Pool{
...
+
+// This flusher is run in a separate goroutine so that the filter can
+// run separately from accepting incoming messages
+func flusher(f filter, incoming chan []Line, done chan struct{}, out pipeline.OutputChannel) {
+	if pdebug.Enabled {
+		g := pdebug.Marker("flusher goroutine")
+		defer g.End()
+	}
+
+	defer close(done)
```
Comment (mattn) [inline filter.go]: is this required?

File: filter.go [hunk 2]
```diff
@@ -215,50 +217,63 @@ var filterBufPool = sync.Pool{
...
-			}
-			releaseRegexpFilterBuf(buf)
-		}
-	}()
+	go flusher(f, flush, flushDone, out)
 
-	buf := getRegexpFilterBuf()
-	defer func() { releaseRegexpFilterBuf(buf) }()
+	buf := getFilterLineBuf()
+	defer func() { releaseFilterLineBuf(buf) }()
```
Comment (mattn) [inline filter.go]: why closure?

File: internal/util/util.go [hunk 3]
```diff
@@ -9,6 +9,12 @@ type fder interface {
 	Fd() uintptr
 }
 
+func CaseInsensitiveIndexFunc(r rune) func(rune) bool {
+	return func(v rune) bool {
```
Comment (mattn) [inline internal/util/util.go]: `lr := unicode.ToUpper(r)` and use to compare below.

File: filter.go (hunk 1 above)
Comment (lestrrat) [inline filter.go]: this one is required. we need to wait until all output has been sent through `out`

File: filter.go (hunk 2 above)
Comment (lestrrat) [inline filter.go]: ほんまや

**Comment by mattn** [comment]
leave to syohex

**Comment by syohex** [comment]
I've checked that fuzzy feature works well.  I'll check code later.

**Comment by syohex** [comment]
Looks good

**Pull request merged**